lib/SPIRV/libSPIRV/SPIRVModule.cpp

```diff
@@ -1232,6 +1232,7 @@ SPIRVInstruction *SPIRVModuleImpl::addIndirectCallInst(
 
 SPIRVInstruction *SPIRVModuleImpl::addFunctionPointerINTELInst(
     SPIRVType *TheType, SPIRVFunction *TheFunction, SPIRVBasicBlock *BB) {
+  TheFunction->addDecorate(DecorationReferencedIndirectlyINTEL);
```

Comment:
Note: this actually changes the semantic:

I guess that sometimes compiler is able to infer which exact function is used and insert direct call instead of an indirect, inline the function and remove its body and declaration from the module. However, with this declaration function must not be removed from the module even if it called directly and inlined.

One more concern is that this decoration is enabled under separate capability, because it is not really needed for function pointers to work, but used to enable different use cases.

Probably the best place for such change is some specific backend and not generic SPIR-V generator, but since the extension is vendor-specific and in state of first public draft, I don't have strong objections against this change right now. Just keep in mind that in longer term this might need to be moved to some specific backend/SPIR-V consumer

Reply:
Fine, I'll drop this